### PR TITLE
Add minimal-versions CI

### PR DESCRIPTION
**Pull request by NobodyXu**
Help discover bugs in version requ
irements

### CODE REVIEW

**Comment by NobodyXu** [comment]
cc @Byron seems like libz-sys is ok?

I'm not sure if I'm running with the right feature set

**Comment by Byron** [comment]
Even though it says it uses minimal versions, I am missing `cargo update` with minimal versions set. Right now it would use Cargo.lock or the latest available, *I think*.

**Comment by NobodyXu** [comment]
Thanks, I was braindead when writing this 😂

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -144,3 +144,22 @@ jobs:
...
           - x86_64-unknown-linux-musl
+
+  minimal-versions:
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          submodules: true
+
+      - run: cargo build
```
Comment (NobodyXu) [inline .github/workflows/ci.yml:155]:
```suggestion
      - run: |
          rustup toolchain install nightly --profile minimal --no-self-update
          cargo +nightly update -Zminimal-versions

      - run: cargo build
```

**Comment by Byron** [comment]
I tried to get the actual `Cargo-zng.toml` file tested as well, but on Windows it's failing as it appears to be unable to find visual studio related environment variables, and I couldn't figure out why.
So I just left it untested, but applied the expected fix anyway.

Once CI is green I can create a new release and then fix `flate2`, which ultimately fixes `gitoxide`.

**Review by Byron — APPROVED** [review]

**Pull request merged**